### PR TITLE
List Recommendations by Source ID

### LINKED ISSUE #26: Need the ability to List recommendations by source product ID
**As a** manager
 **I need** to all recommendations by source product ID
 **So that** customers can see all recommendations from a certain product
   
 ### Details and Assumptions
 * Implement the function of listing all recommendations by source product ID
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given a source product ID
 When a call is made to request all recommendations by that ID
 Then a list of recommendations is returned
 ```

### PR DESCRIPTION
**Pull request by aianurag09**

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -32,7 +32,13 @@ def index():
...
 def list_recommendations():
     """Returns all of the recommendation"""
-    recommendations = Recommendation.all()
+    app.logger.info("Request for recommendations list")
+    recommendations = []
+    src_product_id = request.args.get("src_product_id")
+    if src_product_id:
+        recommendations = Recommendation.find_by_src_id(int(src_product_id))
+    else:
+        recommendations = Recommendation.all()
```
Comment (freshao1017) [inline service/routes.py]: How should we combined src_product_id and rec_product_id together in the same function?

File: tests/test_routes.py [hunk 2]
```diff
@@ -220,6 +221,21 @@ def test_disable_recommendation(self):
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
         disabled = resp.get_json()
         self.assertEqual(disabled["status"], "DISABLED")
+
+    def test_query_pet_recommendations_by_source(self):
```
Comment (changtk07) [inline tests/test_routes.py:225]: what is a pet recommendation?

File: tests/test_models.py [hunk 3]
```diff
@@ -235,6 +235,15 @@ def test_find_by_rec_product_id(self):
     def test_find_or_404_not_found(self):
         """Find or return 404 NOT found"""
         self.assertRaises(NotFound, Recommendation.find_or_404, 0)
+
+    def test_find_by_sourceID(self):
```
Comment (changtk07) [inline tests/test_models.py:239]: naming convention...

File: service/models.py [hunk 4]
```diff
@@ -172,8 +172,24 @@ def init_db(cls, app: Flask):
...
+
+        :param id: the id of the source product to find
+        :type id: int
+
+        :return: list of Recommendations
+        :rtype: list
+
+        """
+        return cls.query.filter(cls.src_product_id == source_id)
+=======
```
Comment (changtk07) [inline service/models.py:189]: run the code before PR.....

**Review by changtk07 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**